src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -269,13 +321,108 @@ public void DeleteIndex([NotNull] string name)
 
         #endregion
 
-        #region IDispose
+        #region Public Methods - QueryFactory
 
-        public void Dispose()
+        public IQuery CreateQuery([NotNull] string queryExpression)
         {
             throw new NotImplementedException();
         }
 
         #endregion
+
+        #region Internal Methods
+
+        internal void GetSpec()
+        {
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4coll == null)
+                    return;
+
+                var spec = Native.c4coll_getSpec(c4coll);
+                Name = spec.name.CreateString();
+                Scope.Name = spec.scope.CreateString();
+            });
+        }
+
+        /// <summary>
+        /// Returns false if this collection has been deleted, or its database closed.
+        /// </summary>
+        internal bool IsCollectionValid()
+        {
+            bool isValid = false;
+            ThreadSafety.DoLocked(() =>
+            {
+                isValid = Native.c4coll_isValid(_c4coll);
+                if (!isValid) {
+                    isValid = Native.c4coll_isValid(_c4coll);
+                }
```

Comment:
This, if intentional, surely deserves a comment for why this is happening.

Reply:
Update the logic.

Reply:
It seems you are checking c4coll_isValid before checking for null, this should be reversed.

Reply:
Logic of this method is updated already. Not sure why "Outdated" is not showing.

Reply:
This could use some consolidation at this point, but it will suffice.